builtin/providers/nsone/resource_zone.go

```diff
@@ -0,0 +1,161 @@
+package nsone
+
+import (
+	"strings"
+
+	"github.com/bobtfish/go-nsone-api"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func zoneResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"zone": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"link": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ttl": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"nx_ttl": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"refresh": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"retry": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"expiry": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"hostmaster": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"dns_servers": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"primary": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+		Create: ZoneCreate,
+		Read:   ZoneRead,
+		Update: ZoneUpdate,
+		Delete: ZoneDelete,
+	}
+}
+
+func zoneToResourceData(d *schema.ResourceData, z *nsone.Zone) {
+	d.SetId(z.Id)
+	d.Set("hostmaster", z.Hostmaster)
+	d.Set("ttl", z.Ttl)
+	d.Set("nx_ttl", z.Nx_ttl)
```

Comment:
sure but those would need changes to go-nsone-api and as mentioned we should may be leave it for laters ?